File: array_and_hashing/TwoSum.java
import java.util.HashMap;
import java.util.Map;

public class TwoSum {

    public static int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> hashmap = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            if (hashmap.containsKey(complement)) {
                return new int[]{i, hashmap.get(complement)};
            }
            hashmap.put(nums[i], i);
        }

        return new int[]{};
    }
}

File: week1-array-string-hashing/two_sum.ts
function twoSum(nums: number[], target: number): number[] {
  const seen: Record<number, number> = {};
  for (let i = 0; i < nums.length; i++) {
    const complement = target - nums[i];
    if (complement in seen) return [i, seen[complement]];
    seen[nums[i]] = i;
  }
  return [];
}
